networking/multiple_networks/understanding-multiple-networks.adoc

```diff
@@ -6,57 +6,56 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-In Kubernetes, container networking is delegated to networking plugins that
-implement the Container Network Interface (CNI).
-
-{product-title} uses the Multus CNI plugin to allow chaining of CNI plugins.
-During cluster installation, you configure your _default_ pod network. The
-default network handles all ordinary network traffic for the cluster. You can
-define an _additional network_ based on the available CNI plugins and attach
-one or more of these networks to your pods. You can define more than one
-additional network for your cluster, depending on your needs. This gives you
-flexibility when you configure pods that deliver network functionality, such as
-switching or routing.
+In Kubernetes, container networking is managed through networking plugins that implement the Container Network Interface (CNI). 
+
+{product-title} uses the Multus CNI plugin to allow chaining of CNI plugins. During cluster installation, you configure your default pod network. The default network handles all ordinary network traffic for the cluster. You can define an additional network based on the available CNI plugins and attach one or more of these networks to your pods. You can define more than one additional network for your cluster, depending on your needs. This gives you flexibility when you configure pods that deliver network functionality, such as switching or routing.
+
+Administrators or users can leverage xref:../../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[user-defined networks (UDNs)] or xref:../../networking/multiple_networks/primary_networks/about-primary-nwt-nad.adoc#understanding-multiple-networks[NetworkAttachmentDefinition (NADs)] to create additional networks of the following types:
```

Comment:
firstly this is way way better than before, awesome job with this refactor

secondly UDNs can be used only when OCP is installed with OVNK, not with other CNIs, NADs are also not implemented by Multus alone needs the chained CNI to do it.. @dougbtv do you have a nice suggestion on how we can include that info here? the paragrah starts with multus CNI but multus only knows NADs not UDNs..

Reply:
@dougbtv @tssurya joe and I tried out hands at better explaining this. Here's our draft paragraph

```suggestion

By default, OVN-Kubernetes serves as the Container Network Interface (CNI) of an {product-title} cluster. With OVN-Kubernetes as the default CNI of a cluster, {product-title} administrators or users can leverage xref:../../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[user-defined networks (UDNs)] or xref:../../networking/multiple_networks/primary_networks/about-primary-nwt-nad.adoc#understanding-multiple-networks[NetworkAttachmentDefinition (NADs)] to create one, or multiple, default networks that handle all ordinary network traffic of the cluster. Both user-defined networks and Network Attachment Definitions can serve as the following network types:

* *Primary networks*: Act as the primary network for the pod. By default, all traffic passes through the primary network unless a route is configured to send traffic through other networks, or a network policy is configured to restrict traffic.

* *Secondary networks*: Act as additional, non-default networks for a pod. Secondary networks provide separate interfaces dedicated to specific traffic types or purposes. Only pod traffic that is explicitly configured to use a secondary network is routed through its interface.

However, during cluster installation, {product-title} administrators can configure alternative default pod networks by leveraging the Multus CNI plugin. With Multus, multiple CNI plugins such as ipvlan, macvlan, or Network Attachment Definitions can be chained together to serve as secondary networks for pods.

[NOTE]
====
User-defined networks are not available for use with the Multus CNI plugin.
====

You can define an additional network based on the available CNI plugins and attach one or more of these networks to your pods. You can define more than one additional network for your cluster depending on your needs. This gives you flexibility when you configure pods that deliver network functionality, such as switching or routing.

For a complete list of supported CNI plugins, see "Additional networks in {product-title}".

For information about user-defined networks, see xref:../../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks (UDNs)].

For information about Network Attachment Definitions, see xref:../../networking/multiple_networks/primary_networks/about-primary-nwt-nad.adoc#understanding-multiple-networks[Creating primary networks using a NetworkAttachmentDefinition].
```

Reply:
@tssurya hows this look?

Reply:
I'm onboard with that! Thanks Steven

Reply:
One small change...

I think this captures the initial portion Surya mentioned...

> With OVN-Kubernetes as the default CNI of a cluster, {product-title} administrators or users can leverage UDN

But, I think this needs a slight change:

> With Multus, multiple CNI plugins such as ipvlan, macvlan, or Network Attachment Definitions can be chained together to serve as secondary networks for pods.

to

> Using Network Attachment Definitions will enable multiple CNI plugins to be executed with Multus CNI, such as ipvlan, macvlan, and bridge CNI to attach to secondary networks for pods.

Reply:
@stevsmit : `For information about user-defined networks, see About user-defined networks (UDNs)` how can audience know which page you have this section in? not intuitive from docs index....

Reply:
@tssurya Addressed here. I can explain indexing later (in short, we rely more on people CTRL+Finding quoted material than i'd like to admit). For now, I have added a link to the section! :) 

Additionally, I've addressed the above sentence to read as follows: 

[NOTE]
====
User-defined networks are only available when OVN-Kubernetes is used as the CNI. They are not supported for use with other CNIs.
====

Thanks for explaining Multus.